File: Compromisso.java
public abstract class Compromisso{
    protected int id;
    protected char tipo;
    protected Data data;
    protected Hora hora;
    protected int duracao;
    protected int prioridade;
    protected int prioridadeFinal;
    protected String descricao;

    public Compromisso(int id, char tipo, Data data, Hora hora, int duracao, int prioridade, String descricao){
        this.id = id;
        this.tipo = tipo;
        this.data = data;
        this.hora = hora;
        this.duracao = duracao;
        this.prioridade = prioridade;
        this.descricao = descricao;
        this.prioridadeFinal = prioridade;
    }

    public abstract void calcularPrioridadeFinal();
    public abstract String getDescricao();
    public abstract boolean isAdiavel();

    public String getIdFormatado(){
        return String.format("%06d", id);
    }

    public int getId(){
        return id;
    }
    
    public char getTipo(){
        return tipo;
    }

    public Data getData(){
        return data;
    }

    public Hora getHora(){
        return hora;
    }

    public int getDuracao(){
        return duracao;
    }

    public int getPrioridade(){
        return prioridade;
    }

    public int getPrioridadeFinal(){
        return prioridadeFinal;
    }
    
    //comparaçao
    public boolean antes(Compromisso outro){
        if (!this.data.equals(outro.data)){
            return this.data.antes(outro.data);
        }
        return this.hora.antes(outro.hora);
    }
    
    public boolean maiorPrioridade(Compromisso outro){
        //compara prioridade final
        return this.prioridadeFinal > outro.prioridadeFinal;
    }
    
    public boolean menorId(Compromisso outro){
        return this.id < outro.id;
    }
    
    public boolean menorDuracao(Compromisso outro){
        return this.duracao < outro.duracao;
    }
}

File: python/compromisso.py
class Compromisso:
    def __init__(self, id, tipo, data, hora, duracao, prioridade, descricao):
        self.id = id
        self.tipo = tipo
        self.data = data
        self.hora = hora
        self.duracao = duracao
        self.prioridade = prioridade
        self.descricao = descricao
        self.prioridadeFinal = prioridade
    
    def calcularPrioridadeFinal(self):
        raise NotImplementedError
    
    def getDescricao(self):
        raise NotImplementedError
    
    def isAdiavel(self):
        raise NotImplementedError
    
    def getIdFormatado(self):
        return f"{self.id:06d}"
    
    def getId(self):
        return self.id
    
    def getTipo(self):
        return self.tipo
    
    def getData(self):
        return self.data
    
    def getHora(self):
        return self.hora
    
    def getDuracao(self):
        return self.duracao
    
    def getPrioridade(self):
        return self.prioridade
    
    def getPrioridadeFinal(self):
        return self.prioridadeFinal
    
    def antes(self, outro):
        if not self.data.iguais(outro.data):
            return self.data.antes(outro.data)
        return self.hora.antes(outro.hora)
    
    def maiorPrioridade(self, outro):
        return self.prioridadeFinal > outro.prioridadeFinal
    
    def menorId(self, outro):
        return self.id < outro.id
    
    def menorDuracao(self, outro):
        return self.duracao < outro.duracao